### PR TITLE
[TS] LPS-167763 Search Bar Scope Everything doesn't return Site members documents results for Guest role if View Document scope isn't also All Sites And Asset Library

### PR DESCRIPTION
**Pull request by georgel-pop-lr**
comes from https://github.com/liferay-search/liferay-portal/pull/647

https://issues.liferay.com/browse/LPS-167763

### CODE REVIEW

**Comment by joshuacords** [comment]
ci:test:relevant

**Comment by joshuacords** [comment]
Hello @georgel-pop-lr,

This produces queries that are too permissible - for example if we add a Site 2 and add documents there but only specify the Default Site and Site 1 in the Role's Permissions, the query will fetch all 6 documents even though only 5 documents should be displayed. (The Site 2 Member's only document is removed during post filtering, but you can try the query through kibana and see that the additional document is fetched.)
Basically the query is saying if any site's permission is given to the Role for an asset type then ignore permissions for that type, but this isn't true.

Let me know if you need further guidance.

**Comment by georgel-pop-lr** [comment]
> Hello @georgel-pop-lr,
> 
> This produces queries that are too permissible - for example if we add a Site 2 and add documents there but only specify the Default Site and Site 1 in the Role's Permissions, the query will fetch all 6 documents even though only 5 documents should be displayed. (The Site 2 Member's only document is removed during post filtering, but you can try the query through kibana and see that the additional document is fetched.) Basically the query is saying if any site's permission is given to the Role for an asset type then ignore permissions for that type, but this isn't true.
> 
> I did like the idea of not changing what is indexed, but that might not be possible.
> 
> Let me know if you need further guidance.

Hi @joshuacords,

Could you please give me some more guidance on how do you see this being fixed?

I tried to add the groupsIds inside searchContext, but it seems that still returns all the documents.

If you need a PTR with diagnose solution, let me know, and I will open one.

Regards.

**Comment by joshuacords** [comment]
@georgel-pop-lr - opening a PTR would be perfect, you can assign it to me once it's created.
